filament/backend/src/metal/MetalHandles.h

```diff
@@ -258,7 +258,9 @@ class MetalRenderTarget : public HwRenderTarget {
     id<MTLTexture> multisampledDepth = nil;
 };
 
-class MetalFence : public HwFence {
+// MetalFence is used to implement both Fences and Syncs.
+// There's no diamond problem, because HwBase (superclass of HwFence and HwSync) is empty.
+class MetalFence : public HwFence, public HwSync {
```

Comment:
maybe add a static_assert for the emptyness of HwBase?

Reply:
Good idea.